Kconfig.zephyr

```diff
@@ -71,6 +71,17 @@ config CODE_DATA_RELOCATION
 	  files should be specified in the CMakeList.txt file with
 	  a cmake API zephyr_code_relocate().
 
+config CODE_DATA_RELOCATION_SRAM
```

Comment:
I believe this is only supported in Cortex-M. Maybe you can make the dependency more strict.

Then: for now, could you move this out of Kconfig.zephyr and place the Kconfig definition in the soc/arm/Kconfig instead? Like this we won't be polluting the main zephyr's Kconfig with SoC-specific Kconfig definitions. I suppose that if this is supported in other ARCHes, later, we can move this Kconfig back here. And you could add the CORTEX_M prefix to the name, to make it more clear (that's optional)

Reply:
`CODE_DATA_RELOCATION` is also only supported in Cortex-M (see a few lines up), so I think we should keep these two config definitions together.

Reply:
OK, that's also fine. Is it possible to move both then, out of Kconfig.zephyr, into Cortex-M arch Kconfig then?

Reply:
Wouldn't it be better if we kept the 2 configs together in the same place.

Reply:
Done